modules/virt-about-aaq-operator.adoc

```diff
@@ -0,0 +1,77 @@
+
+
+[id="aaq-controller-and-crds_{context}]
+== AAQ Operator controller and custom resources
+The AAQ Operator introduces two new API objects defined as custom resource definitions (CRDs) for managing alternative quota implementations across multiple namespaces:
+
+* `ApplicationAwareResourceQuota`: Sets aggregate quota restrictions enforced per namespace. The `ApplicationAwareResourceQuota` API is compatible with the native `ResourceQuota` object and shares the same specification and status definitions.
++
+.Example manifest
+[source,yaml]
+----
+apiVersion: aaq.kubevirt.io/v1alpha1
+kind: ApplicationAwareResourceQuota
+metadata:
+  name: example-resource-quota
+spec:
+  hard:
+    requests.memory: 1Gi 
+    limits.memory: 1Gi
+    requests.cpu/vmi: "1" # <1>
+    requests.memory/vmi: 1Gi # <2>
+# ...
+----
+<1> The maximum amount of CPU that is allowed for VM workloads in the default namespace.
+<2> The maximum amount of RAM that is allowed for VM workloads in the default namespace.
+
+* `ApplicationAwareClusterResourceQuota`: Mirrors the `ApplicationAwareResourceQuota` object at a cluster scope. It is compatible with the native `ClusterResourceQuota` API object and shares the same specification and status definitions. When creating an AAQ cluster quota, you can select multiple namespaces based on annotation selection, label selection, or both by editing the `spec.selector.labels` or `spec.selector.annotations` fields.
++
+.Example manifest
+[source,yaml]
+----
+apiVersion: aaq.kubevirt.io/v1alpha1
+kind: ApplicationAwareClusterResourceQuota # <1>
+metadata:
+  name: example-resource-quota
+spec:
+  quota:
+    hard:
+    requests.memory: 1Gi
+    limits.memory: 1Gi
+    requests.cpu/vmi: "1"
+    requests.memory/vmi: 1Gi
+  selector:
+    annotations: null
+    labels:
+      matchLabels:
+        kubernetes.io/metadata.name: default
+# ...
+----
+<1> You can only create an `ApplicationAwareClusterResourceQuota` object if the `spec.allowApplicationAwareClusterResourceQuota` field in the `HyperConverged` custom resource (CR) is set to `true`.
++
+[NOTE]
+====
+If both `spec.selector.labels` and `spec.selector.annotations` fields are set, only namespaces that match both are selected.
+====
+
+The AAQ controller uses a scheduling gate mechanism to evaluate whether there is enough of a resource available to run a workload. If so, the scheduling gate is removed from the pod and it is considered ready for scheduling. The quota usage status is updated to indicate how much of the quota is used. If the CPU and memory requests and limits for the workload exceed the enforced quota usage limit, the pod is not scheduled and remains in `SchedulingGated` status until there is enough quota available. An error message is displayed with details on why the quota was exceeded.
```

Comment:
```suggestion
The AAQ controller uses a scheduling gate mechanism to evaluate whether there is enough of a resource available to run a workload. If so, the scheduling gate is removed from the pod and it is considered ready for scheduling. The quota usage status is updated to indicate how much of the quota is used. If the CPU and memory requests and limits for the workload exceed the enforced quota usage limit, the pod is not scheduled and remains in `SchedulingGated` status until there is enough quota available. An event with details on why the quota was exceeded will get created.
```

Reply:
Please feel free to use clearer language. I think the current text is a bit misleading. As I understand it, if there isn't enough quota, the pod won't fail right away.
 Instead, the pod will be created but will stay in a "ScheduleGated" status. To find out why it wasn't scheduled, users will need to check the events from the cluster.

Reply:
@Barakmor1 What type of event is created? A `FailedScheduling` event? Will it be helpful to link to [this doc](https://docs.openshift.com/container-platform/4.17/nodes/clusters/nodes-containers-events.html)?

Reply:
it is a Warning, here example: 

```
6s          Warning   Warning            pod/virt-launcher-vm-fedora-4zhqc          exceeded quota: example-arq, requested: requests.cpu/vmi=1, used: requests.cpu/vmi=1, limited: requests.cpu/vmi=1
```

Reply:
The type is warning.
The object is the pod.